### ISSUE TITLE
Cannot deserialize to List<IDictionary<string, object>>

### ISSUE THREAD

**Issue by vkirienko** [dotnet/corefx#42401]
Hi,

Deserializing json to single dictionary works fine but fails if I try to deserialize list of dictionaries. If I remove key with array value it also works.

```
// this works
var payload1 = "{\"CompositeID\":5942,\"Currency\":\"USD\",\"Deal\":[\"Test1\",\"Test92\"]}";
var dict = JsonSerializer.Deserialize<IDictionary<string, object>>(payload1);

// this fails with exception below
var payload2 = "[{\"CompositeID\":5942,\"Currency\":\"USD\",\"Deal\":[\"Test1\",\"Test92\"]}]";
var list = JsonSerializer.Deserialize<List<IDictionary<string, object>>>(payload2);

// this works
var payload3 = "[{\"CompositeID\":5942,\"Currency\":\"USD\"}]";
var list = JsonSerializer.Deserialize<List<IDictionary<string, object>>>(payload3);

```
### Error:
The JSON value could not be converted to System.Collections.Generic.Dictionary`2[System.String,System.Object]. Path: $[0].Deal | LineNumber: 0 | BytePositionInLine: 46.

### Stack trace:
   at System.Text.Json.ThrowHelper.ThrowJsonException(String message, Utf8JsonReader& reader, String path, Exception innerException)
   at System.Text.Json.ThrowHelper.ThrowJsonException_DeserializeUnableToConvertValue(Type propertyType, Utf8JsonReader& reader, String path, Exception innerException)
   at System.Text.Json.ReadStackFrame.CreateEnumerableValue(Utf8JsonReader& reader, ReadStack& state)
   at System.Text.Json.JsonSerializer.HandleStartArray(JsonSerializerOptions options, Utf8JsonReader& reader, ReadStack& state)
   at System.Text.Json.JsonSerializer.ReadCore(JsonSerializerOptions options, Utf8JsonReader& reader, ReadStack& readStack)
   at System.Text.Json.JsonSerializer.ReadCore(Type returnType, JsonSerializerOptions options, Utf8JsonReader& reader)
   at System.Text.Json.JsonSerializer.ParseCore(String json, Type returnType, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.Deserialize[TValue](String json, JsonSerializerOptions options)
   at Deserialization.Program.Main(String[] args) in C:\Projects\Test\Deserialization\Program.cs:line 20

Thank you.

**Comment by layomia** [comment]
This has been fixed for master/5.0 in https://github.com/dotnet/corefx/pull/41482. Can you try the fix in the [latest build of master](https://github.com/dotnet/core-sdk/#installers-and-binaries)?

**Comment by vkirienko** [comment]
I tested it with master/5.0 and can confirm that it works as expected.
